### DISCUSSION TITLE
How to set timer in survey-text?

### DISCUSSION THREAD

**Discussion by Hanqiu0718** [jspsych/jsPsych#2574] [Q&A]
We are asking participants to write a 5-min essay. How do I modify the variable to show this page for only 5 mins, and then will automatically direct to the next page after 5 mins? Here's my code:
```
var LIQ = {
  type: 'survey-text',
  preamble: `<img style="width:5000px;"></img>`,
  questions: [
    {prompt: networkData.LIQ, rows: 20},
  ],
  on_finish: function(data){
      networkData.recallText = JSON.parse(jsPsych.data.get().last().values()[0].responses).Q0; //
      }
}
```

**Comment by jodeleeuw** [comment]
I think automatically advancing to the next page will require modifying the plugin. 

If you want to get close without modifying the plugin, you could use the `on_load` event in the `LIQ` trial to disable the `<button>` that lets the user submit the response, and use `setTimeout` to create a timer that enables the button after 5 minutes.

```js
on_load: () => {
  document.querySelector('button').disabled = true;
  setTimeout( () => { document.querySelector('button').disabled = false }, 1000 * 60 * 5 )
}
```

This will not auto advance the trial but it will make it at least 5 minutes. 

I suppose you could also disable the text input after 5 minutes:

```js
on_load: () => {
  document.querySelector('button').disabled = true;
  setTimeout( () => { 
    document.querySelector('button').disabled = false; 
    document.querySelector('textarea').disabled = true;
  }, 1000 * 60 * 5 )
}
```
